docs/design/conan.md

```diff
@@ -0,0 +1,415 @@
+# Adding Conan Support to Hermeto Should Never Happen
```

Comment:
nitpick: Do we want this strong wording - _...Should Never Happen_? I would just put _[REJECTED]_ in there and the reason being that while it may currently be the case that it's not feasible for us to spend time on this, the trend in various ecosystems is to move towards the direction we'd like so we may just need to wait a few years and I think that should be captured in the decision.